### PR TITLE
feat: New design updates for v2 components

### PR DESCRIPTION
**Pull request by albinAppsmith**
## Description

Design updates after new token implementation.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

- Manual on storybook 
- Manual on main repo
- Jest
- Cypress

### Issues raised during DP testing
https://github.com/appsmithorg/design-system/pull/394#issuecomment-1453044949
https://github.com/appsmithorg/design-system/pull/394#issuecomment-1453235162
https://github.com/appsmithorg/design-system/pull/394#issuecomment-1453260145
https://github.com/appsmithorg/design-system/pull/394#issuecomment-1453298128
https://github.com/appsmithorg/design-system/pull/394#issuecomment-1458122322

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by shadabbuchh** [comment]
### Bugs in the Input component

**Input (textarea):**

- Functional:

1. Can the Label, Error msg & Desc when long not exceed the boundary of the input?
2. `endIcon` & `startIcon` should not be editable when `as` -> `textarea`
3. When positioning the Label on left, no max width is defined if large text is entered

- Design:

1. Wrong padding/gap for the input area/placeholder
2. Wrong background for input area in all states (expected is white)
3. No icon before error message & no red border

___________________________________________________________________________________________________________

**Input (input):**

- Functional:

1. When description/error message is long, the end icon shifts out of the input area.
2. Can the Label, Error msg & Desc when long not exceed the boundary of the input?
3. When positioning the Label on left, no max width is defined if large text is entered
4. When a long text is entered inside the text area, we neither see the complete text nor a horizontal scroll to view the complete text

- Design:

1. Wrong background for input area in all states (expected is white)
2. No icon before error message
3. Wrong gap between label, input area & desc for small variant

**Comment by shadabbuchh** [comment]
### Bug in Button component:

1. Gap between Icon & the text in Medium Button is 4px. Expected is 8px

**Comment by shadabbuchh** [comment]
### Bug in Modal component:

1. Icon button is of 34*34 size.
Screenshot 2023-03-03 at 3 49 25 PM

**Comment by albinAppsmith** [comment]
@shadabbuchh Fixed the issue you have raised... Please reverify.

**Review by tanvibhakta — CHANGES_REQUESTED** [review]

File: packages/design-system/src/Button/Button.styles.tsx [hunk 1]
```diff
@@ -18,37 +18,49 @@ const Variables = css`
...
+    `,
+    md: css`
+      --button-font-weight: 600;
+      --button-font-size: 14px;
+      --button-padding: ${isIconButton
+        ? "var(--ads-v2-spaces-3)"
+        : "var(--ads-v2-spaces-3) var(--ads-v2-spaces-4)"};
+      --button-gap: var(--ads-v2-spaces-3);
+    `,
+    // lg: css`
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Button/Button.styles.tsx:39]: Done

File: packages/design-system/src/Button/Button.styles.tsx [hunk 2]
```diff
@@ -168,18 +181,19 @@ export const StyledButton = styled.button<{
...
   ${({ kind }) => kind && Kinds[kind]}
 
   /* Button heights */
-  ${({ size }) => size && Heights[size]}
+  ${({ isIconButton, size }) => size && getHeights(size, isIconButton)}
 
   position: relative;
   cursor: pointer;
-  border-radius: var(--ads-v2-border-radius);
+  border-radius: var(--ads-v2-border-radius) !important;
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Button/Button.styles.tsx:196]: Radix was overriding this. They have a flag called asChild, when we use that flag this all get neglected. So till they fix that isse we need to make it important.

File: packages/design-system/src/Button/Button.styles.tsx [hunk 3]
```diff
@@ -191,7 +205,7 @@ export const StyledButton = styled.button<{
   padding: 0;
   box-sizing: border-box;
   overflow: hidden;
-  min-width: min-content;
+  min-width: ${({ isIconButton }) => (isIconButton ? "unset" : "isIconButton")};
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Button/Button.styles.tsx:208]: This was a typo. Fixed...

File: packages/design-system/src/Button/Button.tsx [hunk 4]
```diff
@@ -63,22 +65,26 @@ const Button = forwardRef<HTMLButtonElement, ButtonProps>(
...
         {/* Button content */}
-        <ButtonContent className={ButtonContentClassName} size={size}>
+        <ButtonContent
+          className={ButtonContentClassName}
+          isIconButton={isIconButton}
+          size={size}
+        >
           {/* Start Icon Section */}
           {startIcon ? (
             typeof startIcon === "string" ? (
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Button/Button.tsx:80]: No we decided to not do that in icon... You only suggested not to do that

File: packages/design-system/src/Button/Button.types.ts [hunk 5]
```diff
@@ -20,6 +20,8 @@ export type ButtonProps = {
   isLoading?: boolean;
   /** Whether the button is disabled. */
   isDisabled?: boolean;
+  /** Whether the button is a icon button. */
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Button/Button.types.ts:23]: Fixed.

File: packages/design-system/src/Icon/Icon.tsx [hunk 6]
```diff
@@ -4,12 +4,10 @@ import { IconContainer } from "./Icon.styles";
 import { IconProps, IconSizes } from "./Icon.types";
 import { IconProvider } from "./Icon.provider";
 
-const iconSizes: Record<IconSizes, string> = {
+export const iconSizes: Record<IconSizes, string> = {
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Icon/Icon.tsx:7]: This is not a type, this is a value mapping. And this needs to be passed to icon container and provider.

File: packages/design-system/src/Input/Input.styles.tsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+import { InputSizes } from "./Input.types";
+import { Text } from "Text";
+import { iconSizes } from "Icon";
+
+const Variables = css`
+  --input-color-border: var(--ads-v2-color-border);
+  --input-padding-x: var(--ads-v2-spaces-2); // padding left and right
+  --input-padding-y: var(--ads-v2-spaces-2); // padding top and bottom
+  --input-color: var(--ads-v2-color-fg);
+  --input-font-size: 12px;
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Input/Input.styles.tsx:16]: Fixed

File: packages/design-system/src/Input/Input.styles.tsx [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+export const InputSection = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: var(--ads-v2-spaces-2);
+  width: 100%;
+`;
+
+export const InputContainer = styled.div<{
+  disabled?: boolean;
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Input/Input.styles.tsx:79]: Done

File: packages/design-system/src/Toast/Toast.styles.tsx [hunk 9]
```diff
@@ -49,13 +51,14 @@ export const StyledToast = styled(ToastContainer).attrs<ToastProps>({
...
+  align-items: center;
   justify-content: space-between;
+  color: var(--ads-v2-color-fg-emphasis);
+  gap: var(--ads-v2-spaces-3);
 `;
 
 export const StyledButton = styled(Button)`
-  margin-right: -var(--ads-v2-spaces-3);
-  margin-left: var(--ads-v2-spaces-3);
+  min-width: fit-content;
```
[the remark this answers is not in the document]
Comment (albinAppsmith) [reply packages/design-system/src/Toast/Toast.styles.tsx:61]: Done

**Review by tanvibhakta — CHANGES_REQUESTED** [review]
Everything else looks good, but since we've migrated all icons to our current assets folder and updated the names, let's not worry about passing in a custom icon in places that are using an icon.

**Review by tanvibhakta — APPROVED** [review]

**Comment by shadabbuchh** [comment]
### Issues noticed in this PR after the latest commits:

**Input:**
1. The height of the Input -> small variant is incorrect (expected is 22px).
___________________________________________________________________________________________________________

**Callout:**
Functional:

1. How do I add new links using the default input method?
2. When the text is quite long the x icon is not completely visible
Screenshot 2023-03-07 at 6 19 37 PM

Design:

1. When there is a single line, the icon does not seem to be aligned with the text
Screenshot 2023-03-07 at 6 20 13 PM

___________________________________________________________________________________________________________

**Tabs:**
Functional:

1. We are able to enter negative values in the notificationCount control.

Design:

1. Wrong colour for text as well as the bottom line that appears in hover state.
2. Wrong height of the notificationCount.
3. Wrong colour of the horizontal line.

**Comment by albinAppsmith** [comment]
@shadabbuchh Fixed all the issue you mentioned.

- Wrong height of the notificationCount => We are using tag component here so it will follow tag component height
- Can we add an ellipsis when the tab's name is very long along with a tolltip? => We need a clarity from design team. Can do this in next phase. Please add this comment in zeplin.
- When notificationCount is set to 0 should the notif icon be seen? => made it hidden

**Comment by shadabbuchh** [comment]
All identified issues have been fixed and retested.

**Pull request merged**